### PR TITLE
feat(ingest): Add exception metrics to event_manager and post_process_group

### LINKED ISSUE #42178: Make ingestor fault tolerant and production ready
Right now we haven't done a comprehensive overview of metrics and error handling for the ingestor (and associated pipeline steps like `post_process` and `eventstream`). We should:
1. Ensure metric consistency in our new ingestor pipeline with our current processing pipeline
2. Make sure that all new code pathways fail gracefully
    a. Error handling where appropriate
    b. Invalid payloads are rejected with helpful error messages and don't cause unhandled internal errors
3. Set up SLO's for existing event and transaction processing with an eye towards occurrence SLO's

### PR DESCRIPTION
**Pull request by rjo100**
Adds exception metrics to be used in Issue Platform upcoming SLOs

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -448,6 +448,10 @@ def run_post_process_job(job: PostProcessJob):
         try:
             pipeline_step(job)
         except Exception:
+            metrics.incr(
+                "sentry.tasks.post_process.post_process_group.exception",
+                tags={"issue_category": group_event.group.issue_category},
```
Comment (wedamija) [inline src/sentry/tasks/post_process.py:453]: `issue_category` is an enum. You'll want to convert this to either the int value, or maybe better, use the lower case name of the specific enum type.

File: src/sentry/tasks/store.py [hunk 2]
```diff
@@ -708,6 +708,9 @@ def _do_save_event(
             if cache_key:
                 with metrics.timer("tasks.store.do_save_event.delete_cache"):
                     processing.event_processing_store.delete_by_key(cache_key)
+        except Exception:
+            metrics.incr("events.save_event.exception", tags={"event_type": event_type})
+            raise
```
Comment (jjbayer) [inline src/sentry/tasks/store.py:713]: Wouldn't a raised exception here be logged to sentry / sentry4sentry, giving you the error counts for free?

Comment (rjo100) [reply src/sentry/tasks/store.py:713]: So re-raising the exception would move the attribution for the exception? Right now this is just for counting exceptions that are thrown during `eventmanager.save()` pretty much. We wouldn't be handling any exceptions differently other than re-raising them from a different fire

Comment (wedamija) [reply src/sentry/tasks/store.py:713]: I think we specifically want this logged to datadog for use in their SLO feature.

**Review by jjbayer — APPROVED** [review]

File: src/sentry/tasks/store.py (hunk 2 above)
Comment (rjo100) [reply src/sentry/tasks/store.py:713]: Ah I misread the comment. Yeah we want the datadog SLO stuff, not really about the metric itself. Thx!

**Pull request merged**